packages/rocketchat-livechat/server/methods/returnAsInquiry.js

```diff
@@ -1,21 +1,56 @@
 Meteor.methods({
-	'livechat:returnAsInquiry'(rid) {
-		if (!Meteor.userId() || !RocketChat.authz.hasPermission(Meteor.userId(), 'view-l-room')) {
-			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'livechat:saveDepartment' });
+	'livechat:returnAsInquiry'(rid, departmentId) {
+		const room = RocketChat.models.Rooms.findOneById(rid);
+
+		if (!room) {
+			throw new Meteor.Error('error-invalid-room', 'Invalid room', { method: 'livechat:returnAsInquiry' });
+		}
+
+		//When an user goes offline, for example, their user is no longer available
+		//In this case, an error is raised, so to avoid this the room user is instantiated
+		let user = Meteor.user();
+		if (!user) {
+			user = RocketChat.models.Users.findOne(room.servedBy._id);
 		}
 
-		// //delete agent and room subscription
+		if (!user._id || !RocketChat.authz.hasPermission(user._id, 'view-l-room')) {
+			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'livechat:returnAsInquiry' });
+		}
+
+		const agentIds = [];
+		//get the agents of the department
+		if (departmentId) {
+			let agents = RocketChat.Livechat.getOnlineAgents(departmentId);
+
+			if (agents.count() === 0 && RocketChat.settings.get('Livechat_guest_pool_with_no_agents')) {
+				agents = RocketChat.Livechat.getAgents(departmentId);
+			}
+
+			if (agents.count() === 0) {
+				return false;
+			}
+
+			agents.forEach((agent) => {
+				agentIds.push(agent.agentId);
+			});
+		}
+
+		//delete agent and room subscription
 		RocketChat.models.Subscriptions.removeByRoomId(rid);
 
 		// remove user from room
-		const username = Meteor.user().username;
+		const username = user.username;
```

Comment:
this **must** be the agent and not the logged in user.. imagine a situation where the manager is returning the livechat to the queue.